city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,164 @@
+            )
+            self.link_parse_complete = True
+            raise scrapy.exceptions.DontCloseSpider
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+        """
+        for item in response.xpath("//iframe/@src"):
+            yield scrapy.Request(item.get(), callback=self._parse_minutes)
+
+    def _parse_minutes(self, response):
+        """ Traverse tree of URLs and populate self.link_map """
+        for item in response.xpath("//td[@class='name']/a"):
+            try:
+                href = item.xpath("@href")[0].get()
+                text = item.xpath("b/text()")[0].get().strip()
+                if text[-4:] == ".pdf":
+                    text = text[:-4]
+            except IndexError:
+                continue
+
+            url = urljoin("https://{}".format(self.allowed_domains[0]), href)
+            if ".pdf" not in url:
+                # Not a link to meeting minutes file - go a level deeper
+                yield scrapy.Request(url, callback=self._parse_minutes)
+            else:
+                # Dates are given in several formats:
+                format_strs = ["%m-%d-%Y", "%m-%d-%y", "%m/%d/%Y", "%m/%d/%y"]
+                dt = None
+                for format_str in format_strs:
+                    try:
+                        dt = datetime.strptime(text, format_str).date()
+                    except ValueError:
+                        continue
+                    else:
+                        break  # Found a format_str that matches - stop looking
+                if dt is None:
+                    continue  # Could not find a matching format_str - can't process link.
+
+                self.link_map[dt] = url
+
+    def _parse_json(self, response):
+        """ Parse JSON from https://pcb.illinois.gov/ClerksOffice/GetCalendarEvents -> Meetings """
+        data = json.loads(response.body_as_unicode())
+
+        for item in data:
+            if item["CalendarTypeDesc"] == "Holiday":
+                continue  # Not a meeting - just an indication of a holiday on the calendar.
+            title = item["CalendarTypeDesc"].replace("CANCELLED", "").strip()
+            meeting = Meeting(
+                title=title,
+                description="",  # Too inconsistent to parse accurately
+                classification=self._parse_classification(title),
+                start=self._parse_start(item),
+                end=None,
+                all_day=item["IsFullDay"],
+                time_notes="",
+                location=self._parse_location(item),
+                links=list(),
+                source=self._parse_source(item),
+            )
+
+            meeting["links"] = self._parse_links(meeting)
+            meeting["status"] = self._parse_status(meeting, item)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_status(self, meeting, item):
```

Comment:
The `_get_status` method in the `CityScrapersSpider` implements a lot of this behavior, so you can get the same result by removing this method and passing the text as a `kwarg`, so this would work:

```python
meeting["status"] = self._get_status(meeting, text=" ".join([item['CalendarTypeDesc'], item['Description']]).lower())
```

Reply:
Wow, you're right! The embarrassing part is I looked at the `CityScrapersSpider._get_status` method first and somehow reached the conclusion that it wouldn't work. And then proceeded to write something almost identical.